packages/core/src/tools/memoryTool.ts

```
@@ -98,15 +125,27 @@
 
+// Global event emitter for memory operations
+export const memoryEventEmitter = new EventEmitter();
+
+// Optional callback for context engine integration
+export interface MemoryOperationCallbacks {
+  onMemoryAdded?: (event: MemoryAddedEvent) => Promise<void> | void;
+  onMemoryIndexed?: (event: MemoryIndexedEvent) => Promise<void> | void;
+}
+
 export class MemoryTool extends BaseTool<SaveMemoryParams, ToolResult> {
   static readonly Name: string = memoryToolSchemaData.name!;
```

Biome (1.9.4)

[error] Forbidden non-null assertion.



(lint/style/noNonNullAssertion)